python/cudf/cudf/tests/dataset_generator.py

```diff
@@ -200,13 +255,173 @@ def generate(
+                        is_sorted=False,
+                        dtype=dtype,
+                    )
+                )
+            elif dtype.kind in ("U", "O"):
+                column_params.append(
+                    ColumnParameters(
+                        cardinality=cardinality,
+                        null_frequency=null_frequency,
+                        generator=lambda cardinality=cardinality: [
+                            mimesis.random.random.schoice(
+                                string.printable,
+                                meta.get("max_string_length", 1000),
+                            )
+                            for _ in range(cardinality)
+                        ],
+                        is_sorted=False,
+                        dtype=dtype,
+                    )
+                )
+            elif dtype.kind == "M":
+                column_params.append(
+                    ColumnParameters(
+                        cardinality=cardinality,
+                        null_frequency=null_frequency,
+                        generator=datetime_generator(size=cardinality),
+                        is_sorted=False,
+                        dtype=np.dtype(dtype),
+                    )
+                )
+            elif dtype.kind == "m":
+                column_params.append(
+                    ColumnParameters(
+                        cardinality=cardinality,
+                        null_frequency=null_frequency,
+                        generator=timedelta_generator(size=cardinality),
+                        is_sorted=False,
+                        dtype=np.dtype(dtype),
+                    )
+                )
+            elif dtype.kind == "b":
+                column_params.append(
+                    ColumnParameters(
+                        cardinality=cardinality,
+                        null_frequency=null_frequency,
+                        generator=boolean_generator(cardinality),
+                        is_sorted=False,
+                        dtype=np.dtype(dtype),
+                    )
+                )
+            else:
+                raise TypeError(f"Unsupported dtype: {dtype}")
+            # TODO: Add List column support once
+            # https://github.com/rapidsai/cudf/pull/6075
+            # is merged.
+
+    df = get_dataframe(
+        Parameters(num_rows=rows, column_parameters=column_params, seed=seed,),
+        use_threads=True,
+    )
+
+    return df
+
+
+def int_generator(dtype, size):
+    iinfo = np.iinfo(dtype)
+    return lambda: np.random.randint(
+        low=iinfo.min, high=iinfo.max, size=size, dtype=dtype,
+    )
+
+
+def float_generator(dtype, size):
+    finfo = np.finfo(dtype)
+    return (
+        lambda: np.random.uniform(
+            low=finfo.min / 2, high=finfo.max / 2, size=size,
+        )
+        * 2
+    )
+
+
+def datetime_generator(size):
+    return lambda: np.random.randint(low=0, high=2147483647 - 1, size=size,)
```

Comment:
Should use negatives for datetimes as well since that indicates before epoch, should similarly be able to use `np.iinfo` on `int64` to cover a wider range.